### ISSUE TITLE
Use of enforce_idempotency Causes 'Failed to complete #converge action'

### ISSUE THREAD

**Issue by pantocrator27** [test-kitchen/test-kitchen#1225]
## Description

Use of enforce_idempotency Causes 'Failed to complete #converge action'

I have a kitchen.yml file:
```
driver:
  name: vagrant

provisioner:
  name: chef_zero
  deprecations_as_errors: true
  enforce_idempotency: true
  multiple_converge: 2

verifier:
  name: inspec

platforms:
  # https://github.com/chef/bento
  - name: centos-6.8
  - name: centos-7.3
  - name: ubuntu-14.04
  - name: ubuntu-16.04

suites:
  - name: default
    run_list:
      - recipe[cookbook-under-test]
```

when performing a `kitchen converge` I receive:
```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 4 actions failed.
>>>>>>     Failed to complete #converge action: [undefined local variable or method `data' for #<Kitchen::Provisioner::ChefZero:0x007fbcecc20140>] on default-centos-68
>>>>>>     Failed to complete #converge action: [undefined local variable or method `data' for #<Kitchen::Provisioner::ChefZero:0x007fbceedd75b8>] on default-centos-73
>>>>>>     Failed to complete #converge action: [undefined local variable or method `data' for #<Kitchen::Provisioner::ChefZero:0x007fbceedb6a20>] on default-ubuntu-1404
>>>>>>     Failed to complete #converge action: [undefined local variable or method `data' for #<Kitchen::Provisioner::ChefZero:0x007fbceed8fbc8>] on default-ubuntu-1604
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

## Kitchen Version

Tell us which version of test-kitchen you are using (`kitchen --version`).  

```ruby
Test Kitchen version 1.16.0
```

## ChefDK Version

If you are running test-kitchen via ChefDK, `chef --version` will provide additional relevent version details.

```ruby
Chef Development Kit Version: 1.3.40
chef-client version: 12.19.36
delivery version: master (69bfa4a76959a8d093511a90fddd7a1f7e43e354)
berks version: 5.6.4
kitchen version: 1.16.0
```

## Platform Version

Tell us which Operating System distribution and version test-kitchen is running on.

Mac OS X

## Replication Case

Tell us what steps to take to replicate your problem.  See [How to create a Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve)
for information on how to create a good replication case.

## Kitchen Output

The relevant output of the test-kitchen run or a link to a gist of the entire run, if there is one.

The debug output (kitchen -l debug) may be useful, but please link to a gist, or truncate it.

Please do not paste the full run out put here.  If the full output would be useful, link to it in a gist.

```ruby
# Copy-paste your results here
```

## Kitchen Diagnose

Please include the output of `kitchen diagnose --all` or `kitchen diagnose --loader` (if the first failed).  This can be helpful in troubleshooting, so please include it in a gist.

## NOTE:

This repository is for core issues with the Test-Kitchen framework, so if the issue is with a plugin you may be referred to file the issue with the appropriate plugin.

**Comment by pantocrator27** [comment]
just checking on this.

**Comment by MarkGibbons** [comment]
I think one of the commits in #1068 fixes this issue.  I don't have time to separate the commit into a standalone PR.

**Comment by cheeseplus** [comment]
Fixed by #1282
